### PR TITLE
New CVar: `mp_give_c4_frags`

### PR DESCRIPTION
**Pull request by 666JuL**
```cpp
// How many bonuses (frags) will get the player who defused or exploded the bomb.
// 3 - (default behaviour)
//
// Default value: "3"
mp_give_c4_frags 3
```

### CODE REVIEW

**Comment by wopox1337** [comment]
I propose to make not a Boolean switch, but a CVar with the number of frags for the actions of the bomb.

`mp_give_c4_frags "3"`

**Comment by abecee** [comment]
> I propose to make not a Boolean switch, but a CVar with the number of frags for the actions of the bomb.
> 
> ```cpp
> mp_give_c4_frags "3"
> mp_give_c4_frags "-10"
> mp_give_c4_frags "0"
> ```

Yep..this will be more useful then 0 and 1 command. For me i want to change the value's frags with wich number i want.

File: regamedll/dlls/game.cpp [hunk 1]
```diff
@@ -162,6 +162,7 @@ cvar_t allchat                           = { "sv_allchat", "0", 0, 0.0f, nullptr
 cvar_t sv_autobunnyhopping               = { "sv_autobunnyhopping", "0", 0, 0.0f, nullptr };
 cvar_t sv_enablebunnyhopping             = { "sv_enablebunnyhopping", "0", 0, 0.0f, nullptr };
 cvar_t plant_c4_anywhere                 = { "mp_plant_c4_anywhere", "0", 0, 0.0f, nullptr };
+cvar_t give_c4_frags                     = { "mp_give_c4_frags", "3", 0, 0.0f, nullptr };
```
Comment (wopox1337) [inline regamedll/dlls/game.cpp:165]:
```suggestion
cvar_t give_c4_frags                     = { "mp_give_c4_frags", "3", 0, 3.0f, nullptr };
```

File: dist/game.cfg [hunk 2]
```diff
@@ -483,3 +483,9 @@ sv_enablebunnyhopping 0
 //
 // Default value: "0"
 mp_plant_c4_anywhere 0
+
+// When set, players can receive credits for defusing or exploding the bomb.
```
Comment (wopox1337) [inline dist/game.cfg:487]:
```suggestion
// How many bonuses (frags) will get the player who defused or exploded the bomb.
```

File: README.md [hunk 3]
```diff
@@ -104,6 +104,7 @@ This means that plugins that do binary code analysis (Orpheu for example) probab
 | sv_autobunnyhopping                | 0       | 0   | 1            | Players automatically re-jump while holding jump button.<br/>`0` disabled <br/>`1` enabled |
 | sv_enablebunnyhopping              | 0       | 0   | 1            | Allow player speed to exceed maximum running speed.<br/>`0` disabled <br/>`1` enabled |
 | mp_plant_c4_anywhere               | 0       | 0   | 1            | When set, players can plant anywhere, not only in bombsites.<br/>`0` disabled <br/>`1` enabled |
+| mp_give_c4_frags               | 3       | -   | -            | When set, players can receive credits for defusing or exploding the bomb. |
```
Comment (wopox1337) [inline README.md:107]:
```suggestion
| mp_give_c4_frags               | 3       | -   | -            | How many bonuses (frags) will get the player who defused or exploded the bomb. |
```

**Review by wopox1337 — CHANGES_REQUESTED** [review]

**Review by wopox1337 — APPROVED** [review]

**Review by wopox1337 — APPROVED** [review]

File: regamedll/dlls/ggrenade.cpp [hunk 4]
```diff
@@ -1084,8 +1084,12 @@ void CGrenade::__API_HOOK(DefuseBombEnd)(CBasePlayer *pPlayer, bool bDefused)
 			CSGameRules()->m_bBombDefused = true;
 			CSGameRules()->CheckWinConditions();
 
-			// give the defuser credit for defusing the bomb
-			m_pBombDefuser->pev->frags += 3.0f;
+#ifdef REGAMEDLL_ADD
+				m_pBombDefuser->pev->frags += give_c4_frags.value;
```
Comment (wopox1337) [inline regamedll/dlls/ggrenade.cpp:1088]:
```suggestion
				m_pBombDefuser->pev->frags += (int)give_c4_frags.value;
```

**Review by wopox1337 — APPROVED** [review]

**Comment by abecee** [comment]
Will be this added in the next update ?

**Comment by Vaqtincha** [comment]
The name cvar is inaccurate and incomprehensible.

**Comment by wopox1337** [comment]
@Vaqtincha @etojuice 😒
If anyone has suggestions for better naming - happy to accept, I will correct.

**Comment by deprale** [comment]
> and "not a hint about the exploder" If anyone has suggestions for better naming - happy to accept, I will correct.
// How many points(frags) will the player be awarded for completing the objective (defuse/explode c4)
// 3 - (default behaviour)
//
// Default value: "3"
mp_objective_bonus 3

**Comment by wopox1337** [comment]
> mp_objective_bonus 3

Map objective can be different. de_, cs_, es_ and etc...

**Comment by deprale** [comment]
>  > mp_objective_bonus 3
> 
> Map objective can be different. de_, cs_, es_ and etc...
> 
> 
> 

Yes, objective bonus sounds good then no? Because it is universal, only need to change the wording a little bit in the description to not only include c4.

'''Complete the objective - 
Defuse/plant(de)
Save VIP (as)
Save Hostages (cs)'''

I thought about different game modes but didn't check what the actual commit do, I just supposed it changed C4 score behaviour based on the cvar name 😅

**Comment by wopox1337** [comment]
https://github.com/s1lentq/ReGameDLL_CS/blob/d8702df5482d8945dff4ef5005261ffbc8b29343/regamedll/dlls/ggrenade.cpp#L1443

- that PR refers only to bomb bonus

**Pull request merged**